src/solvers/base_solver.cpp

```diff
@@ -34,34 +34,18 @@ BaseSolver::BaseSolver(std::shared_ptr<mfem::ParMesh> mesh, int n, int p) : Base
 void BaseSolver::setEssentialBCs(const std::set<int>& ess_bdr, serac::GeneralCoefficient ess_bdr_coef,
                                  FiniteElementState& state, const int component)
 {
-  auto num_attrs = state_.front()->mesh().bdr_attributes.Max();
-
-  serac::BoundaryCondition bc(ess_bdr_coef, component, ess_bdr, num_attrs);
-
-  for (int attr : ess_bdr) {
-    if (std::any_of(ess_bdr_.cbegin(), ess_bdr_.cend(),
-                    [attr](auto&& existing_bc) { return existing_bc.markers()[attr - 1] == 1; })) {
-      SLIC_WARNING("Multiple definition of essential boundary! Using first definition given.");
-      bc.removeAttr(attr);
-    }
-  }
-
-  bc.setTrueDofs(state);
-
-  ess_bdr_.emplace_back(std::move(bc));
+  bcs_.addEssential(ess_bdr, ess_bdr_coef, state, component);
```

Comment:
I think we can delete this functionality from the base class and just put `bcs_.addEssential(...)` etc. in the individual solvers.

Reply:
This is a step toward better following of [this guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i25-prefer-abstract-classes-as-interfaces-to-class-hierarchies).

Reply:
I was thinking that eventually this would be initialized separately (as in https://github.com/LLNL/serac/issues/118#issuecomment-668805621) but I agree that it can be removed from the base in this PR